src/users/users.gateway.ts

```diff
@@ -5,55 +5,104 @@ import {
   ConnectedSocket,
   WebSocketServer,
   OnGatewayConnection,
+  OnGatewayDisconnect,
+  WsException,
+  BaseWsExceptionFilter,
 } from '@nestjs/websockets';
 import { UsersService } from './users.service';
-import { RegisterUserDto } from './dto/register-user.dto';
-import { Socket } from 'socket.io';
+import { RegisterUserDto as ConnectUserDto } from './dto/register-user.dto';
 import { LoginUserDto } from './dto/login-user.dto';
-import { Logger, UsePipes, ValidationPipe } from '@nestjs/common';
+import {
+  ArgumentsHost,
+  BadRequestException,
+  Catch,
+  Logger,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import { AsyncApiPub, AsyncApiSub } from 'nestjs-asyncapi';
+import { SocketDto } from './dto/socket.dto';
+import { SendJwtDto } from './dto/send-jwt-.dto';
+
+@UsePipes(
+  new ValidationPipe({
+    transform: true,
+    //   exceptionFactory(validationErrors = []) {
```

Comment:
this meant to return websockets exceptions instead of HTTP, but i commented it out to get the errors on service side instead of client side, but this should be on production